target_chains/ethereum/contracts/contracts/entropy/Entropy.sol

```diff
@@ -669,38 +669,43 @@ abstract contract Entropy is IEntropy, EntropyState {
             }
         } else {
             // This case uses the checks-effects-interactions pattern to avoid reentry attacks
-            emit RevealedWithCallback(
-                EntropyStructConverter.toV1Request(req),
-                userRandomNumber,
-                providerRevelation,
-                randomNumber
-            );
-            emit EntropyEventsV2.Revealed(
-                provider,
-                req.requester,
-                req.sequenceNumber,
-                randomNumber,
-                false,
-                bytes(""),
-                0, // gas usage not tracked in the old no-gas-limit flow.
-                bytes("")
-            );
-
+            address callAddress = req.requester;
+            EntropyStructs.Request memory reqV1 = EntropyStructConverter
+                .toV1Request(req);
             clearRequest(provider, sequenceNumber);
+            // WARNING: DO NOT USE req BELOW HERE AS ITS CONTENTS HAS BEEN CLEARED
```

Comment:
referencing req.requester in the prior version of this code wasn't a good idea. (that field doesn't get cleared, but not good to make assumptions about that)